zio-http/src/test/scala/zio/http/internal/middlewares/AuthSpec.scala

```diff
@@ -77,6 +76,15 @@ object AuthSpec extends ZIOHttpSpec with HttpAppTestExtensions {
         } @@ basicAuthContextM).merge.mapZIO(_.body.asString)
         assertZIO(app.runZIO(Request.get(URL.empty).copy(headers = successBasicHeader)))(equalTo("user"))
       },
+      test("Extract username via context with Routes") {
+        val app = {
+          Routes(
+            Method.GET / "context" ->
+              Handler.fromFunction[(AuthContext, Request)] { case (c, _) => Response.text(c.value) },
+          ) @@ basicAuthContextM
+        }.toHttpApp
+        assertZIO(app.runZIO(Request.get(URL.root / "context").copy(headers = successBasicHeader)))(equalTo("user"))
+      },
```

Comment:
I'm adding this test because I try to do exactly the same thing in my code (as it's documented here: https://github.com/zio/zio-http/blob/main/docs/dsl/middleware.md#need-for-middlewares-and-handling-aspects) but the code doesn't compile

As we can see in the CI of this PR, here's the compilation error:
```scala
[error] -- [E172] Type Error: /home/runner/work/zio-http/zio-http/zio-http/src/test/scala/zio/http/internal/middlewares/AuthSpec.scala:83:101 
[error] 83 |              Handler.fromFunction[(AuthContext, Request)] { case (c, _) => Response.text(c.value) },
[error]    |                                                                                                     ^
[error]    |
[error]    |Your request handler is required to accept both parameters Unit, as well as the incoming [[zio.http.Request]].
[error]    |This is true even if you wish to ignore some parameters or the request itself. Try to add missing parameters 
[error]    |until you no longer receive this error message. If all else fails, you can construct a handler manually using 
[error]    |the constructors in the companion object of [[zio.http.Handler]] using the precise types..
[error]    |I found:
[error]    |
[error]    |    zio.http.RequestHandlerInput.apply[Unit,
[error]    |      (zio.http.internal.middlewares.AuthSpec.AuthContext, zio.http.Request)](
[error]    |      zio.Zippable.Zippable2[A, B])
[error]    |
[error]    |But method Zippable2 in trait ZippableLowPriority3 does not match type zio.Zippable.Out[Unit, zio.http.Request,
[error]    |  (zio.http.internal.middlewares.AuthSpec.AuthContext, zio.http.Request)].
```

Reply:
Solved thanks to https://github.com/guizmaii/zio-http/pull/1